src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/ProjectImports/ImportTreeProvider.cs

```diff
@@ -302,6 +302,22 @@ IProjectTree2 SyncNode(IReadOnlyList<IProjectImportSnapshot> imports, IProjectTr
                                     }
 
                                     return result;
+
+                                    // Creates a map from each import's project path to the filename of that project path.
+                                    // Although LINQ's Enumerable.ToDictionary extension method can be used to create such
+                                    // a map, that implementation fails when there are duplicate entries in the imports.
+                                    // Therefore, this implementation loops through the imports to create the map.
+                                    Dictionary<string, string> CreateProjectPathDictionaryFromImports()
```

Comment:
A generalized concept like this would have been great as an additional extension method for IEnumerable that we could use within the project system. Just add a boolean parameter that indicates it should override duplicates and bam, use it anywhere we need to.